test/modfile_parser.jl

```diff
@@ -0,0 +1,125 @@
+#=
+This file reads Dynare .mod files, and extracts the following information:
+  variables
+  exo_variables
+  parameters
+  calibration
+  model
+  initval
+  endval
+  shocks
+=#
+
+using DataStructures
+
+function modfile_parser(mod_file_name)
+  f = open(mod_file_name)
+  lines = readlines(f)  # reads in line by line
+
+
+  # Remove lines beginning with comments
+  filter!(x -> ~ismatch(r"^\s*%", x), lines)
+  filter!(x -> ~ismatch(r"^\s*//", x), lines)
+
+  # Remove lines with comments after text
+  for ln = 1:length(lines)
+    lines[ln] = split(lines[ln], r"%.*$")[1]
+  end
+
+  # Remove lines that are empty
+  filter!(x -> (x != "\n"), lines)
+
+  # Remove new lines
+  for ln = 1:length(lines)
+    lines[ln] = strip(lines[ln])
+  end
+
+  # Smoosh text back together
+  text = join(lines)
+
+
+  # Get variable names
+  tmp = match(r"var (.*?);", text)
+  variables = split(tmp[1], " ")
```

Comment:
I tried this, but using `split(tmp[1], r",?\s*")`  splits all characters (i.e. we get a vector of individual letters..), and `match(r",?\s*", tmp[1])` returns a single string with a space (i.e. `" "`).  I've left it as above, but if you can figure out what might work instead let me know.

Oh and the `\t` issue wasn't showing up at this stage. The `\t` seem to show up when the lines are joined together.